deployment-e2e/molecule/ultimate-commons/oracle-docker-compose.yml

```diff
@@ -6,11 +6,30 @@
   - name: stop the service
     shell: service poabridge stop
 
+  - name: Build current oracle image
+    shell: docker build -t oracle:ultimate-testing --file oracle/Dockerfile --build-arg DOT_ENV_PATH=e2e-commons/components-envs/oracle.env .
```

Comment:
Do you think it is possible in the future to have the step to build the image be part of a separate job in `.circleci/config.yml`? So, overall process could be similar to:
1. Build the docker image
2. Use the docker image in tests
3. Publish the built docker image on the docker hub as intermediate image (the tag of image could be equal the git commit id)
4. If the release is tagged the built docker image will be published with the tag equals to the release version.

The idea is to have exactly the same docker image on the docker hub repo which was used for testing after the git push/merge event.

What do you think?

Reply:
I think it is a good idea. It seems to be possible, here is some documentation on how to build and publish the docker images to docker hub as steps in circleci config https://circleci.com/blog/using-circleci-workflows-to-replicate-docker-hub-automated-builds/
We could open a new issue to address it

Reply:
In fact, we have an issue for this: https://github.com/poanetwork/tokenbridge/issues/100